src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV2.cs

```diff
@@ -289,20 +301,32 @@ private async Task<int> GetPipelineIdAsync(AgentTaskPluginExecutionContext conte
             }
             var definitions = new List<int>() { definition };
 
-            List<Build> list;
-            if (pipelineVersionToDownload == pipelineVersionToDownloadLatest)
-            {
-                list = await buildHttpClient.GetBuildsAsync(project, definitions, tagFilters: tagFilters, queryOrder: BuildQueryOrder.FinishTimeDescending, resultFilter: resultFilter);
-            }
-            else if (pipelineVersionToDownload == pipelineVersionToDownloadLatestFromBranch)
-            {
-                list = await buildHttpClient.GetBuildsAsync(project, definitions, branchName: branchName, tagFilters: tagFilters, queryOrder: BuildQueryOrder.FinishTimeDescending, resultFilter: resultFilter);
-            }
```

Comment:
These two branches were equivalent as `branchName == null` if `pipelineVersionToDownload  == pipelineVersionToDownloadLatest`. Merged just to DRY up the logic as I was adding a third call, but I'm happy to keep it split if you'd prerferm